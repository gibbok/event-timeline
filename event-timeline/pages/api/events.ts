import { ResponseEventsAPI } from '@/api/types'
import { eventsData } from '@/stubs/events'
import type { NextApiRequest, NextApiResponse } from 'next'

const validateNumericQueryParamOrDefault = (defaultValue: number) => (query: string | string[] | undefined): number => {
  const parsedValue = Number(!query ? defaultValue : Array.isArray(query) ? query[0] : query)
  if (Number.isNaN(parsedValue)) {
    throw new Error(`Invalid query parameter: ${query}`)
  }
  return parsedValue
}


const getNumericQueryParamOrDefault = validateNumericQueryParamOrDefault(1)

type ServerError = {
  error: string
}

export default function handler(
  req: NextApiRequest,
  res: NextApiResponse<ResponseEventsAPI | ServerError>
) {
  try {
    const { page: pageQueryParam, limit: limitQueryParam } = req.query

    const page = getNumericQueryParamOrDefault(pageQueryParam)
    const limit = getNumericQueryParamOrDefault(limitQueryParam)

    const start = (page - 1) * limit
    const end = start + limit

    return res.status(200).json({
      countEvents: eventsData.length,
      events: eventsData.slice(start, end)
    })
  } catch (error) {
    if (error instanceof Error) {
      return res.status(500).json({ error: error.message })
    } else {
      return res.status(500).json({ error: 'Unknown error' })
    }
  }
}
