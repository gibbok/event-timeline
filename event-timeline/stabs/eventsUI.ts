import { EventsUI } from "@/components/views/EventTimeline/types";

export const eventDataUI = [
    {
        "eventId": "be317a24-09b8-4b5b-997b-a8542a1738a2",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "downtime_report_submitted",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:40:49.193277+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        }
    },
    {
        "eventId": "e6c2c29d-156d-49ca-9c55-59c6ff464892",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "in_progress",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:40:46.800652+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        }
    },
    {
        "eventId": "87874db1-f5be-4a49-be2f-409c5c3ac4d6",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "headcount_changed",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:33:53.638472+00",
        "jobHeadcount": 4
    },
    {
        "eventId": "68240b61-09d3-4797-9052-7a3f4d3b5223",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "headcount_changed",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:33:42.851223+00",
        "jobHeadcount": 1
    },
    {
        "eventId": "ad29fb9c-d345-431d-9ae6-299ba7f4bcb6",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "submit_downtime_report_causes",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:31:23.379595+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        },
        "causes": {
            "design": [
                "Drawing Issues"
            ],
            "rework": [
                "Drawing not followed"
            ]
        }
    },
    {
        "eventId": "8a6ac826-ed60-45cd-8011-004ca86b5daf",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "submit_downtime_report_causes",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:31:20.127669+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        },
        "causes": {
            "design": [
                "Drawing Issues"
            ]
        }
    },
    {
        "eventId": "d270c3b6-96c3-43d1-ac85-7c512de0be34",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "downtime",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:31:15.617801+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        }
    },
    {
        "eventId": "1666659d-7e5c-43e6-bbc7-c52f8a9a779e",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "in_progress",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:31:10.523076+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        }
    },
    {
        "eventId": "9d0f4e98-11d3-4d02-b361-9cd3ef8b49d9",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "on_break",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:31:04.523876+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        }
    },
    {
        "eventId": "5e884a00-5840-4840-9aa7-ed270933941f",
        "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
        "type": "started",
        "userName": "Thad Cormier",
        "occurrence": "2022-11-23 17:30:59.179995+00",
        "device": {
            "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
            "zoneName": "Bench - Panels",
            "factoryName": "Urban Splash House Factory",
            "partnerName": "Urban Splash",
            "stationName": "Bay 1"
        },
        "jobHeadcount": 2
    }
] as EventsUI