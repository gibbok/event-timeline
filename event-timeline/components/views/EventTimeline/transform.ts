import { CausesAPI, DeviceAPI, EventsAPI, JobEventMetadataAPI, JobEventTypeAPI } from "@/api/types"
import { DeviceUI, CausesUI, EventsUI } from "./types"


const getDeviceUIFromDeviceAPI = (device: DeviceAPI): DeviceUI =>
({
    deviceId: device.device_id,
    zoneName: device.station.zone_name,
    factoryName: device.station.factory_name,
    partnerName: device.station.partner_name,
    stationName: device.station.station_name
})

const getCausesUIFromCausedAPI = (causes: CausesAPI): CausesUI =>
({
    design: causes.Design,
    rework: causes.Rework
})

export const tranformResponseApiToUiData = (data: EventsAPI): EventsUI =>
    data.map(event =>
    ({
        eventId: event.job_event_id,
        cardId: event.job_card_id,
        type: event.job_event_type,
        userName: event.job_event_user,
        occurrence: event.job_event_occurrence,
        device: event.job_event_metadata.device ? getDeviceUIFromDeviceAPI(event.job_event_metadata.device) : undefined,
        causes: event.job_event_metadata.causes ? getCausesUIFromCausedAPI(event.job_event_metadata.causes) : undefined,
        jobHeadcount: event.job_event_metadata.job_headcount
    }))


