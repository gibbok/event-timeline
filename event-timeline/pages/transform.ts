import { CausesAPI, DeviceAPI, EventsAPI, JobEventMetadataAPI, JobEventTypeAPI } from "@/api/types"

export type Reason = string

export type CausesUI = Readonly<{
    design: ReadonlyArray<Reason>
    rework?: ReadonlyArray<Reason>
}>

export type DeviceUI = Readonly<{
    deviceId: string
    zoneName: string,
    factoryName: string,
    partnerName: string,
    stationName: string
}>

export type EventUI = Readonly<{
    eventId: string;
    cardId: string
    type: JobEventTypeAPI,
    userName: string,
    occurrence: string,
    device?: DeviceUI
    causes?: CausesUI
    jobHeadcount?: number
}>

export type EventsUI = ReadonlyArray<EventUI>

const getDeviceUIFromDeviceAPI = (device: DeviceAPI): DeviceUI =>
({
    deviceId: device.device_id,
    zoneName: device.station.zone_name,
    factoryName: device.station.factory_name,
    partnerName: device.station.partner_name,
    stationName: device.station.station_name
})


const getCausesUIFromCausedAPI = (causes: CausesAPI): CausesUI =>
({
    design: causes.Design,
    rework: causes.Rework
})


export const sortEventsByOccurenceDesc = (data: EventsAPI) => [...data].sort((a, b) =>
    Number(new Date(b.job_event_occurrence)) - Number(new Date(a.job_event_occurrence)))


export const tranformResponseApiToUiData = (data: EventsAPI): EventsUI =>
    data.map(event =>
    ({
        eventId: event.job_event_id,
        cardId: event.job_card_id,
        type: event.job_event_type,
        userName: event.job_event_user,
        occurrence: event.job_event_occurrence,
        device: event.job_event_metadata.device ? getDeviceUIFromDeviceAPI(event.job_event_metadata.device) : undefined,
        causes: event.job_event_metadata.causes ? getCausesUIFromCausedAPI(event.job_event_metadata.causes) : undefined,
        jobHeadcount: event.job_event_metadata.job_headcount
    }))


