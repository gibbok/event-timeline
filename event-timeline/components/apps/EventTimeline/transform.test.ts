import { eventsData } from "@/stabs/events";
import { tranformResponseApiToUiData } from "./transform";

describe("DeviceInfo", () => {
    describe("tranformResponseApiToUiData", () => {
        it("should transform server data to data for ui", () => {
            expect(tranformResponseApiToUiData(eventsData)).toMatchSnapshot()
        });
    });
});