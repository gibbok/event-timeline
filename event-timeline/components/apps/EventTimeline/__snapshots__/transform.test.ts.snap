// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceInfo tranformResponseApiToUiData should transform server data to data for ui 1`] = `
[
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "5e884a00-5840-4840-9aa7-ed270933941f",
    "jobHeadcount": 2,
    "occurrence": "2022-11-23 17:30:59.179995+00",
    "type": "started",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "9d0f4e98-11d3-4d02-b361-9cd3ef8b49d9",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:31:04.523876+00",
    "type": "on_break",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "1666659d-7e5c-43e6-bbc7-c52f8a9a779e",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:31:10.523076+00",
    "type": "in_progress",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "d270c3b6-96c3-43d1-ac85-7c512de0be34",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:31:15.617801+00",
    "type": "downtime",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": {
      "design": [
        "Drawing Issues",
      ],
      "rework": undefined,
    },
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "8a6ac826-ed60-45cd-8011-004ca86b5daf",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:31:20.127669+00",
    "type": "submit_downtime_report_causes",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": {
      "design": [
        "Drawing Issues",
      ],
      "rework": [
        "Drawing not followed",
      ],
    },
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "ad29fb9c-d345-431d-9ae6-299ba7f4bcb6",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:31:23.379595+00",
    "type": "submit_downtime_report_causes",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": undefined,
    "eventId": "68240b61-09d3-4797-9052-7a3f4d3b5223",
    "jobHeadcount": 1,
    "occurrence": "2022-11-23 17:33:42.851223+00",
    "type": "headcount_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": undefined,
    "eventId": "87874db1-f5be-4a49-be2f-409c5c3ac4d6",
    "jobHeadcount": 4,
    "occurrence": "2022-11-23 17:33:53.638472+00",
    "type": "headcount_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "e6c2c29d-156d-49ca-9c55-59c6ff464892",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:40:46.800652+00",
    "type": "in_progress",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "be317a24-09b8-4b5b-997b-a8542a1738a2",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:40:49.193277+00",
    "type": "downtime_report_submitted",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "208782fc-8b3b-4762-88e3-ca18b13e4444",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:40:55.572262+00",
    "type": "on_break",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "427e4d0d-12a4-40ce-902f-6bf330623906",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:41:01.008387+00",
    "type": "clocked_out",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "56e4ab73-27a5-4eba-be87-c8d99f1fdbe0",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:41:04.983673+00",
    "type": "in_progress",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "b41034ee-668e-429f-8cc1-01a75a7bc86d",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:41:07.944163+00",
    "type": "downtime",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": {
      "design": [
        "Coordination",
        "Drawing Issues",
      ],
      "rework": undefined,
    },
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "0f128978-4c91-40d3-8c4d-06f2a806439b",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:41:11.311781+00",
    "type": "submit_downtime_report_causes",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "dfe5d80e-3fc9-4c85-a885-c0d997d6b49d",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:42:49.03318+00",
    "type": "in_progress",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "1d21eec9-b7c9-4023-b697-dab2db5bad6e",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:42:51.176651+00",
    "type": "downtime_report_submitted",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": undefined,
    "eventId": "58ce9443-4389-464a-8258-74c663c0eb16",
    "jobHeadcount": 4,
    "occurrence": "2022-11-23 17:43:02.574084+00",
    "type": "headcount_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "69afaa10-7a07-44a5-b623-554dad59f6f1",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:43:23.158537+00",
    "type": "in_process_check_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "fe1cd8cd-eb42-49e8-92a5-80996e32e345",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:43:28.643534+00",
    "type": "in_process_check_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "058c7edc-de1e-4d62-bfbd-95e66c0dc7ff",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:43:41.745791+00",
    "type": "in_process_check_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "6b694a30-2e72-4626-9ea5-9e0e7eefb662",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:45:00.960674+00",
    "type": "in_process_check_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "9fdd6228-7694-4ad7-944e-0158ac292177",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-23 17:45:17.294235+00",
    "type": "in_process_check_changed",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "d64c1cd5-9134-4913-9e72-3b5fe77cd408",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-25 14:38:33.069967+00",
    "type": "canceled",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "5d4a0bb3-552e-446b-9f6b-f00a88d8a782",
    "jobHeadcount": 1,
    "occurrence": "2022-11-25 14:39:19.751762+00",
    "type": "started",
    "userName": "Thad Cormier",
  },
  {
    "cardId": "ce6bff8d-eb09-4208-9405-c2594f74ccb0",
    "causes": undefined,
    "device": {
      "deviceId": "d85922780c45ad4eadc0919e75f964085db7649b2dcb7fd78b76f8d65ce09cd3",
      "factoryName": "Urban Splash House Factory",
      "partnerName": "Urban Splash",
      "stationName": "Bay 1",
      "zoneName": "Bench - Panels",
    },
    "eventId": "437293ac-e12c-439a-8511-007da10900c4",
    "jobHeadcount": undefined,
    "occurrence": "2022-11-25 14:39:25.149178+00",
    "type": "canceled",
    "userName": "Thad Cormier",
  },
]
`;
